scripts/ensure-venv.sh

```diff
@@ -1,21 +1,26 @@
 #!/bin/bash
-set -xe
 
 # optionally opt out of virtualenv creation
 if [ "x$SENTRY_NO_VIRTUALENV_CREATION" == "x1" ]; then
-  exit 0
+    exit 0
 fi
 
-# if we already have a venv, we can leave
-if .venv/bin/python --version &> /dev/null; then
-  exit 0
+if [ -n "$VIRTUAL_ENV" ]; then
+    # we're enforcing that virtualenv be in .venv, since future tooling e.g. venv-update will rely on this.
+    if [ "$VIRTUAL_ENV" != "${PWD}/.venv" ]; then
+        echo "You're in a virtualenv, but it's not in the expected location (${PWD}/.venv)"
+        exit 1
+    fi
+    # TODO: when direnv lands, make the check strictly match .python-version.
+    if ! python -c "import sys; sys.exit(sys.version_info[:2] != (2, 7))"; then
```

Comment:
You know you can do `python --version`, right?

Reply:
I know. Neither is really better than the other though.

Reply:
Reading `python --version` is quite a bit easier. Since there won't be any more python 2.7 version, we can do a direct string comparison with the contents of the `.python-version` file. If you want to be fancy afterwards, you may use this one-liner: https://github.com/getsentry/onpremise/blob/6538f472888bf0f8796faea649a9b32ca80ba959/install.sh#L38